Test/git-issues/git-issue-330.dfy

```diff
@@ -0,0 +1,19 @@
+// RUN: %dafny /compile:0 "%s" > "%t"
+// RUN: %diff "%s.expect" "%t"
+
+
+datatype Option<A> = Some(value: A) | None
+
+datatype T<A> = T(value: Option<A>)
+
+function f(t: T<int>) : T<int>
+{
+  match t.value {
+    case Some(val) => T(Some(val))
+    case None => (
+      var none : Option<int> := None;
+      T(none)
```

Comment:
The bug report listed this as the workaround, not the bug. The bug was that it was not possible to write just `T(None)` for L14-15. It seems that bug is still present.

Reply:
Fixed the test, and included the workarounds as additional tests